### ISSUE TITLE
Why is the accuracy not reproduced on the oxfordpets and ucf101 datasets

### ISSUE THREAD

**Issue by GuanyuLi** [tonyhuang2022/UPL#3]
The reported accuracies on the oxfordpets and ucf101 datasets are 85.83 and 62.70, respectively, but our reproduced accuracies are only 82.91 and 61.19. How can I improve the accuracy?
Here is my environment info:
dassl 0a2acf5c49a0a4131d7fdd44a7b024a2f4d6209a
torch 1.12.0
torchaudio 0.7.2
torchvision 0.13.0
pytorch-cpu 1.1.0
pillow 9.2.0
numpy 1.21.6

**Comment by tonyhuang2022** [comment]
Thanks.
Do you try ensemble training? or only train a prompt?

**Comment by GuanyuLi** [comment]
Thanks for your reply, we only train a prompt, and the reported results of CLIP and UPL on oxfordpets, ucf101 and sun397 datasets are 85.83/88.28, 62.70/67.17 and 60.18/63.98, respectively, but our reproduced results are 82.91/88.03, 61.19/65.69 and 59.78/63.18, respectively. In addition, the performance of other datasets consistent with the paper in my experiments.

**Comment by tonyhuang2022** [comment]
Thanks for the detailed info. I suggest that:
**1. ensure the consistent of train/test datasets,
2. try to update the version of dassl.**

Actually, there is already someone told me that updating the version of dassl can work for the problem.
However, our experiments are all based on the version posted in readme.
So far, I do not know what causes the problem.

**Comment by GuanyuLi** [comment]
Thank you, we have tried many versions of dassl, including the latest version and the posted version in readme, but still haven't solved the problem. In addition, do all the results of UPL in Table 1 use the same training parameters?

**Comment by tonyhuang2022** [comment]
Yes, we only train the prompt. Do you follow the data structure proposed by CoOp?

**Comment by GuanyuLi** [comment]
Yes, we make sure that we follow the data structure proposed by CoOp.
